examples/define-syntax-rule.kl

```diff
@@ -1,44 +1,40 @@
 #lang "prelude.kl"
 
+(import (rename (shift "prelude.kl" 1) [syntax-case raw-syntax-case]))
 (import (shift "prelude.kl" 1))
 (import (shift "do.kl" 1))
 (import (shift "list-syntax.kl" 1))
-(import (shift "quasiquote.kl" 1))
+(import (shift "dot-dot-dot.kl" 1))
 (import (shift "syntax.kl" 1))
 
 (define-macros
-  ((define-macro
+  ([define-macro
      (lambda (stx)
-       (syntax-case stx
-         ((list (_ pattern body))
-          (syntax-case pattern
-           ((cons macro-name args)
-            (pure `(define-macros
-                     ((,macro-name (lambda (stx)
-                                     (syntax-case stx
-                                      ((list ,pattern)
-                                       ,body)))))))))))))))
+       (syntax-case stx ()
+         [(_ (,macro-name ,args ...) ,body)
+          (pure `(define-macros
+                   ([,macro-name
+                     (lambda (stx)
+                       (raw-syntax-case stx
```

Comment:
I'm still using `raw-syntax-case` here so that I don't have to generate a variant of `args` in which every argument is wrapped in a `quote`. I guess that's one small downside of `fancy-syntax-case`'s unorthodox syntax.